### PR TITLE
Jest

### PR DESCRIPTION
**Pull request by rkennel**
## Overview
Change unit testing framework from Jasmine to Jest.  Save 75% of time executing unit tests

Connects #274 

## Testing Instructions
Run the following commands and observe results
```
yarn unit
```
```
yarn unit-coverage
```
```
yarn unit-watch
```

## Note
4 of the test suites had no tests.  These were deleted.

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -46,23 +47,27 @@
...
     "codelyzer": "^5.2.1",
     "jasmine-core": "^3.6.0",
     "jasmine-spec-reporter": "~4.2.1",
+    "jest": "^26.6.3",
+    "jest-preset-angular": "^8.4.0",
     "karma": "~4.4.0",
     "karma-chrome-launcher": "~3.1.0",
     "karma-coverage-istanbul-reporter": "^2.1.1",
     "karma-jasmine": "~3.0.0",
     "karma-jasmine-html-reporter": "~1.5.0",
```
Comment (adandach) [inline ui/package.json:69]:
```suggestion
```
Karma can be safely removed

**Review by adandach — CHANGES_REQUESTED** [review]
Awesome improvement. Definitely prefer jest over jasmine. 

I requested some changes inline. In addition, it looks like this PR introduces yarn? Please also update the README and `build.gradle` to use yarn.

Not required for approval: There are a lot of `@ts-ignore`s to deal with the use of `jest.fn`. Is this how we want to handle using `jest.fn` with TypeScript or is there a different approach we should take? Open for discussion.

File: ui/src/app/modules/teams/services/board.service.spec.ts [hunk 2]
```diff
@@ -17,28 +17,32 @@
 
 import {BoardService} from './board.service';
 import {HttpClient} from '@angular/common/http';
-import {Subject} from 'rxjs';
-import * as moment from 'moment';
+import {Observable, Subject} from 'rxjs';
```
[the remark this answers is not in the document]
Comment (adandach) [reply ui/src/app/modules/teams/services/board.service.spec.ts:20]: Probably better to just clean up the imports across all modified files.

**Comment by rkennel** [comment]
> Awesome improvement. Definitely prefer jest over jasmine.
> 
> Required changes are commented below.
> 
> Not required for approval: There are a lot of `@ts-ignore`s to deal with the use of `jest.fn`. Is this how we want to handle using `jest.fn` with TypeScript or is there a different approach we should take? Open for discussion.

We could setup TS Lint such that it ignores unit tests.  The challenge is that jest.fn does some "magic" that does not play well with types

**Comment by rkennel** [comment]
All requested changes have been made

TSLINT feedback will be added as a new issue

File: ui/src/app/modules/teams/services/save-checker.service.ts [hunk 3]
```diff
@@ -29,6 +29,8 @@ export class SaveCheckerService {
   }
 
   public updateTimestamp(): void {
-    this._lastSavedDateTime = moment().format('h:mm a');
+    const now = new Date();
+    const ampm = now.getHours() > 11 ? 'pm' : 'am';
+    this._lastSavedDateTime = `${now.getHours()}:${now.getMinutes().toString().padStart(2, '0')} ${ampm}`;
```
[the remark this answers is not in the document]
Comment (rkennel) [reply ui/src/app/modules/teams/services/save-checker.service.ts]: Reverted Change now uses moment again

File: ui/src/app/modules/teams/services/websocket.service.spec.ts [hunk 4]
```diff
@@ -28,11 +28,17 @@ describe('WebsocketService', () => {
...
-    spiedClient = createSpyObj({
-      connect: null,
-      subscribe: {messages: new Observable()},
-      send: null
-    });
+    // spiedClient = createSpyObj({
+    //   connect: null,
+    //   subscribe: {messages: new Observable()},
+    //   send: null
+    // });
```
[the remark this answers is not in the document]
Comment (rkennel) [reply ui/src/app/modules/teams/services/websocket.service.spec.ts]: Done

**Review by adandach — APPROVED** [review]

**Pull request merged**